src/providers/oracle/qgsoracleprovider.cpp

```diff
@@ -3393,23 +3393,29 @@ bool QgsOracleProviderMetadata::styleExists( const QString &uri, const QString &
   else if ( qry.value( 0 ).toInt() == 0 )
   {
     // layer styles table does not exist
+    conn->disconnect();
     return false;
   }
 
-  if ( !qry.prepare( QStringLiteral( "SELECT id,stylename FROM layer_styles"
-                                     " WHERE f_table_catalog=?"
-                                     " AND f_table_schema=?"
-                                     " AND f_table_name=?"
-                                     " AND f_geometry_column=?"
-                                     " AND styleName=?" ) ) ||
-       !(
-         qry.addBindValue( dsUri.database() ),
-         qry.addBindValue( dsUri.schema() ),
-         qry.addBindValue( dsUri.table() ),
-         qry.addBindValue( dsUri.geometryColumn() ),
-         qry.addBindValue( styleId.isEmpty() ? dsUri.table() : styleId ),
-         qry.exec()
-       ) )
+  if ( !qry.prepare( "SELECT id,stylename FROM layer_styles"
+                     " WHERE f_table_catalog=?"
+                     " AND f_table_schema=?"
+                     " AND f_table_name=?"
+                     " AND f_geometry_column" +
+                     QString( dsUri.geometryColumn().isNull() ? " IS NULL" : "=?" ) +
```

Comment:
```suggestion
                     QString( dsUri.geometryColumn().isEmpty() ? " IS NULL" : "=?" ) +
```

Null strings are a historical oddity